src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+                if (this->unk_29C == 0) {
+                    this->unk_26C[7] = this->unk_26C[5] = 10000.0f;
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+    if (this->unk_2E0 != this->unk_26C[9]) {
+        Math_SmoothScaleMaxF(&this->unk_2E0, this->unk_26C[9], 0.5f, 4000.0f);
+    }
+
+    if (this->unk_2DC != this->unk_26C[0]) {
+        Math_SmoothScaleMaxF(&this->unk_2DC, this->unk_26C[0], 0.5f, 4000.0f);
+    }
+
+    if (this->unk_2C4 != this->unk_26C[2]) {
+        Math_SmoothScaleMaxF(&this->unk_2C4, this->unk_26C[2], 0.8f, 7000.0f);
+    }
+
+    if (this->unk_2C8 != this->unk_26C[7]) {
+        Math_SmoothScaleMaxF(&this->unk_2C8, this->unk_26C[7], 0.8f, 7000.0f);
+    }
+
+    if (this->unk_2CC != this->unk_26C[8]) {
+        Math_SmoothScaleMaxF(&this->unk_2CC, this->unk_26C[8], 0.8f, 7000.0f);
+    }
+
+    if (this->unk_2D0 != this->unk_26C[1]) {
+        Math_SmoothScaleMaxF(&this->unk_2D0, this->unk_26C[1], 0.8f, 7000.0f);
+    }
+
+    if (this->unk_2D4 != this->unk_26C[5]) {
+        Math_SmoothScaleMaxF(&this->unk_2D4, this->unk_26C[5], 0.8f, 7000.0f);
+    }
+
+    if (this->unk_2D8 != this->unk_26C[6]) {
+        Math_SmoothScaleMaxF(&this->unk_2D8, this->unk_26C[6], 0.8f, 7000.0f);
+    }
+}
+
+void EnNiw_SpawnAttackCucco(EnNiw* this, GlobalContext* globalCtx) {
+    f32 viewX;
+    f32 viewY;
+    f32 viewZ;
+    Vec3f attackCuccoPos;
+    EnAttackNiw* attackCucco;
+
+    if (this->timer5 == 0) {
+        if (this->unk_296 < 7) {
+            viewX = globalCtx->view.lookAt.x - globalCtx->view.eye.x;
+            viewY = globalCtx->view.lookAt.y - globalCtx->view.eye.y;
+            viewZ = globalCtx->view.lookAt.z - globalCtx->view.eye.z;
+            VEC_SET(attackCuccoPos, ((Math_Rand_ZeroOne() - 0.5f) * viewX) + globalCtx->view.eye.x,
+                    Math_Rand_CenteredFloat(0.3f) + ((globalCtx->view.eye.y + 50.0f) + (viewY * 0.5f)),
+                    ((Math_Rand_ZeroOne() - 0.5f) * viewZ) + globalCtx->view.eye.z);
+            attackCucco =
+                Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_ATTACK_NIW, attackCuccoPos.x,
+                                   attackCuccoPos.y, attackCuccoPos.z, 0x0000, 0x0000, 0x0000, 0x0000);
+
+            if (attackCucco != NULL) {
+                this->unk_296++;
+                this->timer5 = 10;
+            } else {
+                osSyncPrintf("\n\n");
+                osSyncPrintf("\x1b[32m ☆☆☆☆☆ 発生できず  ☆☆☆☆☆ \n\x1b[m");
+            }
+        }
+    }
+}
+
+void func_80AB6100(EnNiw* this, GlobalContext* globalCtx, s32 arg2) {
+    f32 phi_f2;
+    f32 new_var;
+
+    if (this->timer4 == 0) {
+        this->timer4 = 3;
+        if ((this->actor.bgCheckFlags & 1) != 0) {
+            this->actor.velocity.y = 3.5f;
+        }
+    }
+
+    if (this->timer3 == 0) {
+        this->unk_2A0++;
+        this->unk_2A0 &= 1;
```

Comment:
```suggestion
        this->unk_2A0 %= 2;
```
may or may not match?

Reply:
Doesn't match, also what does the % mean?

Reply:
`%` is the modulus operator, when used with positive numbers it's the remainder in the Euclidean division
eg `3 % 2 == 1`, `115 % 50 == 15`

with negative numbers I forgot the details. It's not the remainder in the Euclidean division though.